scipy/signal/filter_design.py

```diff
@@ -329,21 +343,47 @@ def freqz(b, a=1, worN=None, whole=False, plot=None):
 
     """
     b, a = map(atleast_1d, (b, a))
-    if whole:
-        lastpoint = 2 * pi
-    else:
-        lastpoint = pi
-    if worN is None:
-        N = 512
-        w = numpy.linspace(0, lastpoint, N, endpoint=False)
-    elif isinstance(worN, int):
-        N = worN
-        w = numpy.linspace(0, lastpoint, N, endpoint=False)
+
+    worN = 512 if worN is None else worN  # historically we allowed None
```

Comment:
I agree it's not the cleanest here. We could emit a warning, but it seemed like overkill to me since supporting the old (now undocumented) behavior takes so little effort.